File: Codigos Examen l prog/Pascal/ContarVocalesConsonantes.pas
program ContarVocalesConsonantes;

var
  cadena: string;
  i: integer;
  longitud: integer;
  vocalCount: integer;
  consonantCount: integer;
  caracter: char;

begin
  writeln('Introduce una cadena:');
  readln(cadena);

  longitud := length(cadena);
  vocalCount := 0;
  consonantCount := 0;

  for i := 1 to longitud do
  begin
    caracter := upcase(cadena[i]); // Convertir a mayúsculas para simplificar la comparación

    if (caracter >= 'A') and (caracter <= 'Z') then
    begin
      if (caracter = 'A') or (caracter = 'E') or (caracter = 'I') or (caracter = 'O') or (caracter = 'U') then
      begin
        vocalCount := vocalCount + 1;
      end
      else
      begin
        consonantCount := consonantCount + 1;
      end;
    end;
  end;

  writeln('Número de vocales: ', vocalCount);
  writeln('Número de consonantes: ', consonantCount);
  readln; // Espera a que el usuario presione Enter para cerrar la consola
end.

File: Codigos Examen l prog/C++/ContarVocalesConsonantes.cpp
#include <iostream>
#include <string>
#include <cctype> // Para la función toupper

int main() {
    std::string cadena;
    int vocalCount = 0;
    int consonantCount = 0;

    std::cout << "Introduce una cadena: ";
    std::getline(std::cin, cadena);

    for (char caracter : cadena) {
        caracter = std::toupper(caracter); // Convertir a mayúsculas para simplificar la comparación

        if (caracter >= 'A' && caracter <= 'Z') {
            if (caracter == 'A' || caracter == 'E' || caracter == 'I' || caracter == 'O' || caracter == 'U') {
                vocalCount++;
            } else {
                consonantCount++;
            }
        }
    }

    std::cout << "Número de vocales: " << vocalCount << std::endl;
    std::cout << "Número de consonantes: " << consonantCount << std::endl;

    return 0;
}